tests/infra/device_connector.py

```diff
@@ -0,0 +1,110 @@
+# SPDX-FileCopyrightText: (c) 2024 Tenstorrent AI ULC
+#
+# SPDX-License-Identifier: Apache-2.0
+
+import os
+from enum import Enum
+from typing import Sequence
+
+import jax
+import jax._src.xla_bridge as xb
+
+
+class DeviceType(Enum):
+    CPU = "cpu"
+    TT = "tt"
+    GPU = "gpu"
+
+
+class DeviceConnector:
+    """
+    Singleton class providing connections to devices on which jax commands will be
+    executed.
+
+    As a singleton it is instantiated only once, thus making sure that PJRT plugin is
+    registered exactly once. Registration needs to happen before any other jax commands
+    are executed. Registering it multiple times would cause error.
+
+    Do not instantiate this class. Use provided global instance.
```

Comment:
I think there are some established ways to create a singleton class in python which protect against this, and which avoid having a global variable for instance. Can you please explore those options?